cdap-ui/app/cdap/components/DataPrep/Directives/Calculate/index.js

```diff
@@ -308,20 +465,46 @@ export default class Calculate extends Component {
     this.setState(newState);
   }
 
+  parseColumns = () => {
+    let columns = typeof this.props.column === 'string' ? [this.props.column] : this.props.column;
+    
+    let type1 = DataPrepStore.getState().dataprep.typesCheck[columns[0]];
+
+    if (columns.length === 1) {
+      return [false, false, columns[0], type1];
```

Comment:
Can this return an object instead of an array? The keys would be self-documenting then.